Task1_3.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+
+
+namespace taskOnePartThree
+{
+    internal class Program
+    {
+        static void Main(string[] args)            
+        {
+            Console.WriteLine("please enter your name");
+            string userName = Console.ReadLine();
+
+            Console.WriteLine($"Hello! {userName}, \n  In order for me to generate your array, first you need to indicate how many elements your array will contain, then you must enter the number that each element must contain one at a time.");
+            Console.Write("\nHow many elements do you want your array to have?: ");
+            
+            if (int.TryParse(Console.ReadLine(), out int userArray) && userArray > 0)
+            {
+                int[] originalUserArray = new int[userArray];
+                int uniqueCount = 0;
+                int[] newUniqueArray = new int[userArray];
+
+                for (int i = 0; i < userArray; i++)
+                {
+                    
+                    Console.Write($"Enter element {i + 1}: ");
+                    if (int.TryParse(Console.ReadLine(), out int userElementInput))
+                    {
+                        originalUserArray[i] = userElementInput;
+                        if (!ContainsNumber(newUniqueArray, uniqueCount, userElementInput))
+                        {
+                            newUniqueArray[uniqueCount] = userElementInput;
+                            uniqueCount++;
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine("Invalid input. Please enter a valid integer.");
+                        i--;
+                    }
+                }
+
+                Console.WriteLine("\nthis is your original array:");
+                PrintArray(originalUserArray);
+
```

Comment:
Cool, but better to create a new array of size uniqueCount and copy values to it from newUniqueArray, its more reliable due to no necessity to store uniqueCount as real array length

Reply:
Array.Copy created